plugin/rewrite/rewrite_test.go

```diff
@@ -20,6 +20,14 @@ func msgPrinter(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, err
 	return 0, nil
 }
 
+func handlerRequestSaver(m *dns.Msg) plugin.HandlerFunc {
```

Comment:
a whole handler to return the message 'm', why is this needed?

Reply:
It is for UT. 
I need to get the DNS request sent AFTER process by Rewrite in order to test that the OPT we modify o that DNS request is correct.